sources/LogglyDestination.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  LogglyDestination.swift
+//  SwiftyBeaver
+//
+//  Created by Francesco Pretelli on 2/03/16.
+//  Copyright © 2016 Sebastian Kreutzberger. All rights reserved.
+//
+
+import Foundation
+private let logglyQueue: dispatch_queue_t = dispatch_queue_create("loggly", DISPATCH_QUEUE_SERIAL)
+
+public class LogglyDestination:BaseDestination {
+    let LOGGLY_URL = "https://logs-01.loggly.com/inputs/"
+    
+    public var logglyToken:String = "insert-token-here"
+    public var logglyTag:String = "http"
+    public var maxEntriesBeforeSend = 1
+    
+    private var buffer:[String] = [String]()
+    
+    public override init() {
+        super.init()
+        self.colored = false
+    }
+    
+    // append to file. uses full base class functionality
+    override public func send(level: SwiftyBeaver.Level, msg: String, thread: String, path: String, function: String, line: Int) -> String? {
+        let formattedString = super.send(level, msg: msg, thread: thread, path: path, function: function, line: line)
+        
+        if let str = formattedString {
+            addLogMsgToBuffer(str)
+        }
+        return formattedString
+    }
+    
+    private func getLogglyUrl() -> String {
+        return LOGGLY_URL + logglyToken + "/tag/" + logglyTag
+    }
+    
+    private func addLogMsgToBuffer(msg:String) {
+        dispatch_async(logglyQueue) {
+            self.buffer.append(msg)
+            if self.buffer.count > self.maxEntriesBeforeSend {
+                let tmpbuffer = self.buffer
+                self.buffer = [String]()
+                self.sendLogsInBuffer(tmpbuffer)
+            }
+        }
+    }
+    
+    private func sendLogsInBuffer(stringbuffer:[String]) {
+        let allMessagesString = stringbuffer.joinWithSeparator("\n")
+        
+        if let allMessagesData = (allMessagesString as NSString).dataUsingEncoding(NSUTF8StringEncoding) {
+            let urlRequest = NSMutableURLRequest(URL: NSURL(string: getLogglyUrl())!)
+            urlRequest.HTTPMethod = "POST"
+            urlRequest.HTTPBody = allMessagesData
+            NSURLConnection.sendAsynchronousRequest(urlRequest, queue: NSOperationQueue(), completionHandler: {
+                (response: NSURLResponse?, responsedata: NSData?, error: NSError?) -> Void in
+                if let anError = error {
+                    print("Error from Loggly: \(anError)")
+                } else {
+                    if let data = responsedata {
+                        print("Posted to Loggly, status = \(NSString(data: data, encoding:NSUTF8StringEncoding))")
+                    }
+                }
+            })
+        }
+    }
```

Comment:
Flush tries to make sure queue is emptied before main thread ends. Probably want any buffered loggly items sent to loggly as well as part of flush.


On Mar 1, 2016, 10:45 PM -0500, Francesco Pretelli<notifications@github.com>, wrote:
>  
> Insources/LogglyDestination.swift(https://github.com/SwiftyBeaver/SwiftyBeaver/pull/67#discussion_r54675807):
>  
> >+    if let allMessagesData = (allMessagesString as NSString).dataUsingEncoding(NSUTF8StringEncoding) {>+      let urlRequest = NSMutableURLRequest(URL: NSURL(string: getLogglyUrl())!)>+      urlRequest.HTTPMethod = "POST">+      urlRequest.HTTPBody = allMessagesData>+      NSURLConnection.sendAsynchronousRequest(urlRequest, queue: NSOperationQueue(), completionHandler: {>+        (response: NSURLResponse?, responsedata: NSData?, error: NSError?) ->Void in>+        if let anError = error {>+          print("Error from Loggly: \(anError)")>+        } else {>+          if let data = responsedata {>+            print("Posted to Loggly, status = \(NSString(data: data, encoding:NSUTF8StringEncoding))")>+          }>+        }>+      })>+    }>+  }
>  
> Not sure I understood it correctly. If you check the buffer is cleared in addLogMsgToBuffer when calling sendLogsInBuffer, and "stringbuffer" is just a local variable that should clear itself once the function returns.
>  
> I've just noticed that if there is an error (loggly not working or anything), the logs are lost so I will implement this in a better way.
>  
>  
> —
> Reply to this email directly orview it on GitHub(https://github.com/SwiftyBeaver/SwiftyBeaver/pull/67/files#r54675807).
>  
>  
>